fuzzers/005-tilegrid/generate_full.py

```diff
@@ -339,7 +339,7 @@ def propagate_IOI_SING(database, tiles_by_grid):
             prev_tile = tile
             tile = tiles_by_grid[(
                 database[tile]['grid_x'], database[tile]['grid_y'] + 1)]
-            if '_SING' in database[tile]['type']:
+            if 'IOI3_SING' in database[tile]['type']:
```

Comment:
This will break on parts that the _SING tiles are not IOI3.  Maybe need IOI18_SING/IOI25_SING/IOI3_SING?

Reply:
I wasn't aware there were IOI18 or IOI25 tiles. Is it the same for IOBs? Does it mean we will need the tilegrid to include all these types?

Reply:
A7 doesn't have them, but K7 does, see:
https://github.com/SymbiFlow/prjxray-db/blob/master/kintex7/site_type_IOB18.json
https://github.com/SymbiFlow/prjxray-db/blob/master/kintex7/tile_type_RIOI.json